src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+      renderWithNewValidations({ id: mockId, [validationType]: "bar" });
+
+      expect(screen.getByText("bar")).toHaveAttribute(
+        "id",
+        `${mockId}-validation`
+      );
+      expect(screen.getByRole("textbox")).toHaveAttribute(
+        "aria-describedby",
+        `${mockId}-validation`
+      );
+    }
+  );
+
+  it("ignores the labelInline and related styling props", () => {
+    renderWithNewValidations({
+      labelInline: true,
+      label: "example label",
+      labelAlign: "left",
+      labelWidth: 100,
+      labelSpacing: 1,
+    });
+    const labelContainer = screen.getByTestId("label-container");
+    expect(labelContainer).toHaveStyle({ width: undefined });
+    expect(labelContainer).toHaveStyle({ justifyContent: undefined });
+    expect(labelContainer).toHaveStyle({ paddingLeft: undefined });
+    expect(labelContainer).toHaveStyle({ paddingRight: undefined });
+  });
+
+  it("renders the hint text with the correct styling when the labelHelp prop is passed", () => {
+    renderWithNewValidations({ labelHelp: "Example hint text" });
+    const hintText = screen.getByText("Example hint text");
+    expect(hintText).toBeInTheDocument();
+    expect(hintText).toHaveStyle({
+      fontSize: "14px",
+      marginTop: "0px",
+      marginBottom: "8px",
+    });
+    expect(hintText).toHaveStyleRule("color", "var(--colorsUtilityYin055)");
+  });
+});
+
+test("accepts ref as a ref object", () => {
+  const ref = { current: null };
+  render(<Textarea ref={ref} />);
+
+  expect(ref.current).toBe(screen.getByRole("textbox"));
+});
+
+test("accepts ref as a ref callback", () => {
+  const ref = jest.fn();
+  render(<Textarea ref={ref} />);
+
+  expect(ref).toHaveBeenCalledWith(screen.getByRole("textbox"));
+});
+
+test("sets ref to empty after unmount", () => {
+  const ref = { current: null };
+  const { unmount } = render(<Textarea ref={ref} />);
+
+  unmount();
+
+  expect(ref.current).toBe(null);
+});
+
+test("renders with the expected default border radius styling", () => {
+  render(<Textarea />);
+  expect(screen.getByRole("textbox")).toHaveStyleRule(
+    "border-radius",
+    "var(--borderRadius050)"
+  );
+});
+
+test("renders with the expected custom border radius styling", () => {
+  render(<Textarea borderRadius="borderRadius200" />);
+  expect(screen.getByRole("textbox")).toHaveStyleRule(
+    "border-radius",
+    "var(--borderRadius200)"
+  );
+});
+
+test("renders with the expected custom border radius styling as an array", () => {
+  render(
+    <Textarea
+      borderRadius={[
+        "borderRadius050",
+        "borderRadius100",
+        "borderRadius200",
+        "borderRadius400",
+      ]}
+    />
+  );
+  expect(screen.getByRole("textbox")).toHaveStyleRule(
+    "border-radius",
+    "var(--borderRadius050) var(--borderRadius100) var(--borderRadius200) var(--borderRadius400)"
+  );
+});
+
+test("fires a console warning if more than four border-radius values are passed", () => {
```

Comment:
comment: if you do end up removing style tests I think this one is worth keeping as we're obviously testing that the warning fires correctly etc